UnitsNet/UnitSystem.cs

```diff
@@ -88,5 +114,55 @@ public override int GetHashCode()
         /// Gets the SI unit system.
         /// </summary>
         public static UnitSystem SI { get; } = new UnitSystem(SIBaseUnits);
+
+        /// <summary>
+        ///     Gets the default unit information for the given quantity type associated with the current unit system.
+        ///     For example: the default unit of length in SI is the 'meter' abbreviated with the letter 'm'.
+        ///     It is possible to customize the associations by creating derived UnitSystems (immutable) using the method
+        ///     <see cref="WithDefaultUnit" />.
+        /// </summary>
+        /// <param name="quantityType">The quantity type of interest.</param>
+        /// <returns>
+        ///     The default UnitInfo for the given quantity type, if such an association exists,
+        ///     and <see langword="null" /> otherwise.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///     Quantity type can not be undefined.
+        /// </exception>
+        public UnitInfo GetDefaultUnitInfo(QuantityType quantityType)
+        {
+            if (quantityType == QuantityType.Undefined)
+                throw new ArgumentException("Quantity type can not be undefined.", nameof(quantityType));
+            return _defaultUnits.Value[(int)quantityType - 1];
+        }
+
+        /// <summary>
+        ///     Create a derived unit system by specifying a default unit for a given quantity type.
+        ///     It is possible to configure multiple associations by chaining calls to this method:
+        ///     SI.WithDefaultUnit(..).WithDefaultUnit(..)...
+        /// </summary>
+        /// <param name="quantityType">The quantity type of interest.</param>
+        /// <param name="defaultUnitInfo">The default UnitInfo to associate with the given quantity type.</param>
+        /// <param name="baseUnits">Optionally provide a new definition for the base units of the new unit system.</param>
+        /// <returns>
+        ///     A new UnitSystem that defines <paramref name="defaultUnitInfo" /> as the default unit for
+        ///     <paramref name="quantityType" />
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///     Quantity type can not be undefined.
+        /// </exception>
+        public UnitSystem WithDefaultUnit(QuantityType quantityType, UnitInfo defaultUnitInfo, BaseUnits baseUnits = null)
+        {
+            if (quantityType == QuantityType.Undefined)
+                throw new ArgumentException("Quantity type can not be undefined.", nameof(quantityType));
+
+            // TODO any way to check if UnitInfo is of QuantityType?
```

Comment:
The two ways I can think of:
1. Add `QuantityType` field to `UnitInfo` so we can read it, but it feels unnecessary
2. Check that unit info exists in `Quantity.GetInfo(QuantityType.Length).UnitInfos`.

Reply:
used option 2